docs: add Windows User Credentials section to Credential-overview (#2466)
dsccommunity/SqlServerDsc: source/WikiSource/Credential-overview.md

then please submit a [new issue](https://github.com/dsccommunity/SqlServerDsc/issues/new?template=Problem_with_resource.md).
Then the community can work together to support built-in accounts for that
DSC resource too.

## Windows User Credentials

When using Windows user credentials (i.e., `LoginType = 'WindowsUser'`) with
SqlServerDsc resources, there are important considerations regarding the
username format and how it is extracted from the `PSCredential` object.

### Username Property vs GetNetworkCredential()

The `PSCredential` object has two ways to access the username:

| Property | Behavior |
|----------|----------|
| `$Credential.UserName` | Returns the **full username as provided**, including domain prefix (NetBIOS or FQDN). |
| `$Credential.GetNetworkCredential().UserName` | Returns **only the username portion**, stripping the domain prefix when using NetBIOS format (`DOMAIN\user` → `user`). |

**Always use `$Credential.UserName`** when passing the username to SMO connection
contexts (e.g., `ConnectAsUserName`). This preserves the domain format exactly

> as the user entered it.

Reviewer: This is referencing properties of SMO types. For examples it should instead reference parameters of the commands or DSC resources. The reader will want to know how to use the commands and DSC resource in this module, not SMO.

### Supported Username Formats

| Format | Example | Works with SQLPS (SQL 2016) | Works with SqlServer module |
|--------|---------|-----------------------------|----------------------------|
| **FQDN (UPN)** | `user@domain.local` | ✅ Yes | ✅ Yes |
| **Username only** | `user` | ✅ Yes | ✅ Yes |
| **NetBIOS** | `DOMAIN\user` | ❌ **No** | ✅ Yes |

> **Important**: The NetBIOS format (`DOMAIN\user`) **does not work** with the
> legacy SQLPS module (SQL Server 2016 and earlier). When using SQLPS,
> you must use either FQDN format (`user@domain.local`) or just the username
> without domain prefix.

This limitation was identified in [issue #1223](https://github.com/dsccommunity/SqlServerDsc/issues/1223)

> where `GetNetworkCredential().UserName` was being used internally, causing

Reviewer: `GetNetworkCredential().UserName` is not used in any code, so this paragraph is incorrect and states past-tense which we do not need to document. Also, if this is still an issue it is not because of `GetNetworkCredential().UserName` and if this is kept it should instead state why it doesn't work today. 🤔

NetBIOS domain names to be stripped and authentication to fail on SQL Server 2016.

### Recommendations

1. **Prefer FQDN format** (`user@domain.local`) for maximum compatibility across
   all SQL Server versions and PowerShell modules.
2. **Avoid NetBIOS format** (`DOMAIN\user`) when targeting SQL Server 2016 or
   environments where SQLPS may be used.
3. **Use `$Credential.UserName`** (not `GetNetworkCredential().UserName`) in

>    any custom code that passes credentials to SMO connection contexts.

Reviewer: ```suggestion
   any custom code that passes credentials to commands or DSC resources.
```

### Example: Creating a Windows User Credential

```powershell
# FQDN format (recommended - works everywhere)
$password = ConvertTo-SecureString 'P@ssw0rd1' -AsPlainText -Force
$username = 'user@company.local'

> $credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $username, $password

Reviewer: We could also do with the short form (could be another example)
```suggestion
$credential = [PSCredential]::new($username, $password) 
```

# Username only (works everywhere)
$username = 'user'

> $credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $username, $password

Reviewer: Set the password in this example too so it more clear, otherwise "username only" could be seen as a `PSCredential` without password, which is not possible.

# NetBIOS format (AVOID for SQL 2016 / SQLPS)
$username = 'COMPANY\user'
$credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $username, $password

> ```

Reviewer: I don't think we need to document how to NOT do it? It is already stated above, so we don't need an example for it?

When using these credentials with SqlServerDsc resources, pass them via the

> built-in `PsDscRunAsCredential` parameter:

Reviewer: Also add something like 

```suggestion
built-in `PsDscRunAsCredential`, credential or password parameters. This example shows the built-in `PsDscRunAsCredential` parameter:
```

```powershell
Configuration Example
{
    param
    (
        [Parameter(Mandatory = $true)]
        [System.Management.Automation.PSCredential]
        $SqlAdministratorCredential
    )

    Import-DscResource -ModuleName 'SqlServerDsc'

    node localhost
    {
        SqlLogin 'Add_WindowsUser'
        {
            Ensure               = 'Present'
            Name                 = 'CONTOSO\WindowsUser'
            LoginType            = 'WindowsUser'
            ServerName           = 'TestServer.company.local'
            InstanceName         = 'DSC'
            PsDscRunAsCredential = $SqlAdministratorCredential
        }
    }
}
```

<sup>_This section was informed by [issue #1223](https://github.com/dsccommunity/SqlServerDsc/issues/1223)

> and [issue #1768](https://github.com/dsccommunity/SqlServerDsc/issues/1768)._</sup>

Reviewer: We don't need to mention issue #1768 as that is what this PR resolves.

```suggestion
<sup>_This section was informed by [issue #1223](https://github.com/dsccommunity/SqlServerDsc/issues/1223)._</sup>
```
